Index: Source/DataStructureAndEncodingDefinition/gdcmReader.cxx
===================================================================
--- Source/DataStructureAndEncodingDefinition/gdcmReader.cxx	(revision 6152)
+++ Source/DataStructureAndEncodingDefinition/gdcmReader.cxx	(working copy)
@@ -264,7 +264,7 @@
   if( F->GetHeader().IsEmpty() )
     {
     hasmetaheader = false;
-    gdcmWarningMacro( "no file meta info found" );
+//    gdcmWarningMacro( "no file meta info found" );
     }
 
   const TransferSyntax &ts = F->GetHeader().GetDataSetTransferSyntax();
