From a1046a9e09f46e0a364dceef061ef4aa9576548f Mon Sep 17 00:00:00 2001
From: "Daniele E. Domenichelli" <daniele.domenichelli@gmail.com>
Date: Wed, 27 Jun 2012 09:43:02 +0200
Subject: [PATCH] Fix warnings on .desktop file reported by lintian

---
 data/qnifti2dicom.desktop |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/data/qnifti2dicom.desktop b/data/qnifti2dicom.desktop
index 468d837..9b57dd6 100644
--- a/data/qnifti2dicom.desktop
+++ b/data/qnifti2dicom.desktop
@@ -1,8 +1,7 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=QNifti2Dicom
 Comment=Convert 3D NIfTI images to DICOM 2D series
 Exec=qnifti2dicom
 Icon=qnifti2dicom
 Type=Application
-Categories=Science;
+Categories=Qt;Education;Science;FileTools;ImageProcessing;MedicalSoftware;
-- 
1.7.10

