--- a/debian/control
+++ b/debian/control
@@ -5,7 +5,7 @@ Maintainer: Daniele E. Domenichelli <daniele.domenichelli@gmail.com>
 Build-Depends: debhelper (>= 7.3.16),
  cdbs, cmake (>= 2.6.4), libqt4-dev (>= 4.4),
  libinsighttoolkit4-dev (>= 4.3.0) [i386 amd64] | libinsighttoolkit3-dev,
- libfftw3-dev,
+ libfftw3-dev, libgdcm2-dev
  libtclap-dev (>= 1.2.0),
  libvtk6-dev | libvtk5-qt4-dev
 Standards-Version: 3.9.5
--- /dev/null
+++ b/debian/patches/series
@@ -0,0 +1 @@
+nd-backport_itk-use-file.patch
